packages/beacon-node/src/network/peers/discover.ts

```diff
@@ -260,100 +258,53 @@ export class PeerDiscovery {
    */
   private onDiscoveredPeer = async (evt: CustomEvent<PeerInfo>): Promise<void> => {
     const {id, multiaddrs} = evt.detail;
-    const status = await this.handleDiscoveredPeer(id, multiaddrs[0]);
+    const attnets = zeroAttnets;
+    const syncnets = zeroSyncnets;
+    const status = await this.handleDiscoveredPeer(id, multiaddrs[0], attnets, syncnets);
     this.metrics?.discovery.discoveredStatus.inc({status});
   };
-  /**
-   * Progressively called by libp2p peer discovery as a result of any query.
-   */
-  private async handleDiscoveredPeer(peerId: PeerId, multiaddrTCP: Multiaddr): Promise<DiscoveredPeerStatus> {
-    try {
-      // Check if peer is not banned or disconnected
-      if (this.peerRpcScores.getScoreState(peerId) !== ScoreState.Healthy) {
-        return DiscoveredPeerStatus.bad_score;
-      }
-
-      // Ignore connected peers. TODO: Is this check necessary?
-      if (this.isPeerConnected(peerId.toString())) {
-        return DiscoveredPeerStatus.already_connected;
-      }
-
-      // Ignore dialing peers
-      if (getDefaultDialer(this.libp2p).pendingDials.has(peerId.toString())) {
-        return DiscoveredPeerStatus.already_dialing;
-      }
-
-      const attnets = zeroAttnets;
-      const syncnets = zeroSyncnets;
-
-      // Should dial peer?
-      const cachedPeer: CachedENR = {
-        peerId,
-        multiaddrTCP,
-        subnets: {attnets, syncnets},
-        addedUnixMs: Date.now(),
-      };
-
-      // Only dial peer if necessary
-      if (this.shouldDialPeer(cachedPeer)) {
-        void this.dialPeer(cachedPeer);
-        return DiscoveredPeerStatus.attempt_dial;
-      } else {
-        // Add to pending good peers with a last seen time
-        this.cachedENRs.set(peerId.toString(), cachedPeer);
-        const dropped = pruneSetToMax(this.cachedENRs, MAX_CACHED_ENRS);
-        // If the cache was already full, count the peer as dropped
-        return dropped > 0 ? DiscoveredPeerStatus.dropped : DiscoveredPeerStatus.cached;
-      }
-    } catch (e) {
-      this.logger.error("Error onDiscovered", {}, e as Error);
-      return DiscoveredPeerStatus.error;
-    }
-  }
 
   /**
    * Progressively called by discv5 as a result of any query.
    */
   private onDiscoveredENR = async (enr: ENR): Promise<void> => {
-    const status = await this.handleDiscoveredENR(enr);
+    if (this.randomNodeQuery.code === QueryStatusCode.Active) {
+      this.randomNodeQuery.count++;
+    }
+    // async due to some crypto that's no longer necessary
+    const peerId = await enr.peerId();
+    // tcp multiaddr is known to be be present, checked inside the worker
+    const multiaddrTCP = enr.getLocationMultiaddr(ENRKey.tcp);
+    if (!multiaddrTCP) {
+      this.logger.error("Discv5 worker sent enr without tcp multiaddr", {enr: enr.encodeTxt()});
+      this.metrics?.discovery.discoveredStatus.inc({status: DiscoveredPeerStatus.error});
+      return;
+    }
```

Comment:
Can we handle this case inside the worker? so the worker always sent fully verified ENRs.

Reply:
This case is handled inside the worker. This condition is included here just for type safety